diff --git a/dist/node/chunk-3d7P_gGE.js b/dist/node/chunk-3d7P_gGE.js
index 2970286cac137f7c15d531954cca600407fdbedd..21f4d4d6fa0955a58904ea1811db3fbc6ecbece1 100644
--- a/dist/node/chunk-3d7P_gGE.js
+++ b/dist/node/chunk-3d7P_gGE.js
@@ -36416,11 +36416,12 @@ function preWrapperPlugin(md, options) {
   md.renderer.rules.fence = (...args) => {
     const [tokens, idx] = args;
     const token = tokens[idx];
+    const title = extractTitle(token.info);
     token.info = token.info.replace(/\[.*\]/, "");
     const active = / active( |$)/.test(token.info) ? " active" : "";
     token.info = token.info.replace(/ active$/, "").replace(/ active /, " ");
     const lang = extractLang(token.info);
-    return `<div class="language-${lang}${active}"><button title="${options.codeCopyButtonTitle}" class="copy"></button><span class="lang">${lang}</span>` + fence(...args) + "</div>";
+    return `<div name="${title}" class="language-${lang}${active}"><button title="${options.codeCopyButtonTitle}" class="copy"></button><span class="lang">${lang}</span>` + fence(...args) + "</div>"
   };
 }
 function extractTitle(info, html = false) {
diff --git a/types/default-theme.d.ts b/types/default-theme.d.ts
index 338c9432c7c7d9ca0eb43bc003a6bbe0fc308dca..3927463060057ae5af75073a80c38f66ca6fa7fc 100644
--- a/types/default-theme.d.ts
+++ b/types/default-theme.d.ts
@@ -268,6 +268,11 @@ export namespace DefaultTheme {
      */
     docFooterText?: string
 
+    /**
+     * Icon for the item
+     */
+    icon?: string
+
     rel?: string
     target?: string
   }
